opsdroid/database/mongo/__init__.py

```diff
@@ -89,4 +99,7 @@ async def delete(self, key):
 
         """
         _LOGGER.debug("Deleting %s from MongoDB.", key)
-        return await self.database[key].delete_one({"$query": {}})
+        if self.query_data:
+            return await self.database[key].delete_one(self.query_data)
+        else:
+            return await self.database.drop_collection(key)
```

Comment:
if `query_data` exists on the object, then use it for the delete, otherwise delete the whole collection (which includes my fix, since the previous code didn't work.)